programs/dibio.c

```diff
@@ -341,6 +341,11 @@ int DiB_trainFromFiles(const char* dictFileName, size_t maxDictSize,
         /* Limit the size of the training data to 2GB */
         /* TODO: there is opportunity to stop DiB_fileStats() early when the data limit is reached */
         loadedSize = (size_t)MIN( MIN((S64)maxMem, fs.totalSizeToLoad), MAX_SAMPLES_SIZE );
+        if (memLimit != 0) {
+            DISPLAYLEVEL(2, "!  Warning : setting manual memory limit for dictionary training data at %u MB \n",
+                (unsigned)(memLimit / (1 MB)));
```

Comment:
Tricky side effect:
if `loadedSize < memLimit`, the reported value might be incorrect.
Prefer updating `loadedSize` first, and then use the value to print the report.

Reply:
My reasoning was that it would be confusing for the user to report `loadedSize` as the limit. For example, suppose the total size of dictionary samples is 1 MB and the user sets memLimit at 10 MB. Then it may be confusing to see a message saying "the manual memory limit has been set at 1 MB".

Reply:
ah ok, you focus on the manual setting itself, rather than the effective limit.
Yes, that's valid.